client/client.go

```diff
@@ -686,10 +685,7 @@ func (r *NotaryRepository) publish(cl changelist.Changelist) error {
 		return err
 	}
 
-	remote, err := getRemoteStore(r.baseURL, r.gun, r.roundTrip)
-	if err != nil {
-		return err
-	}
+	remote := r.remoteStore
```

Comment:
Wait never mind, ignore me - I think we use `OfflineStore` in this case.  Ok, that was why I think we used `getRemoteStore` everywhere - if there was an error, we return `OfflineStore` + the error, and in some cases, we fail the operation if there was an error.

In the case of downloading new data, we ignore the error and just use the `OfflineStore` because we can maybe read from cache.

This means that https://github.com/docker/notary/pull/1094/files#diff-1cd41a2611da49ce63630d5a51bd82c5R78 may not be the right behavior.  However, if we don't propagate the error, we'd lose it.  

Every operation from `OfflineStore` will result in a generic offline error, though, so wondering if we should update `OfflineStore` to take an error it can wrap, and propagate the original error that way?

cc @endophage